src/get_imports.rs

```diff
@@ -7,23 +7,36 @@ use swc_common::source_map::Pos;
 use swc_ecma_ast::Str;
 use swc_ecma_parser::Capturing;
 use swc_ecma_parser::{lexer::Lexer, Parser, StringInput, Syntax};
-pub struct SourceSpecifiers {
+
+use crate::error::GetImportError;
+struct SourceSpecifiers {
     specifiers: Vec<swc_ecma_ast::ImportSpecifier>,
     source: Str,
 }
 
 pub fn get_imports_map_from_file<'a>(
     file_path: &'a PathBuf,
-) -> HashMap<String, Option<HashSet<String>>> {
-    let imports = get_imports_from_file(&file_path);
-    get_imports_map(&imports, &file_path)
+) -> Result<HashMap<String, Option<HashSet<String>>>, GetImportError> {
+    if file_path.exists() {
+        let imports = match get_imports_from_file(&file_path) {
+            Ok(i) => i,
+            Err(e) => return Err(e),
+        };
+        return get_imports_map(&imports, &file_path);
+    }
+    Err(GetImportError::FileDoesNotExist(file_path.to_str().unwrap().to_string()))
 }
 
-fn get_imports_from_file<'a>(file_path: &'a PathBuf) -> Vec<SourceSpecifiers> {
+fn get_imports_from_file<'a>(file_path: &'a PathBuf) -> Result<Vec<SourceSpecifiers>, GetImportError> {
+    let path_string = match file_path.to_str() {
+        Some(path) => path,
+        None => return Err(GetImportError::ReadTsFileError(None)),
+    };
     let cm = Arc::<swc_common::SourceMap>::default();
-    let fm = cm
-        .load_file(Path::new(file_path.to_str().unwrap()))
-        .expect("Could not load file");
+    let fm = match cm.load_file(Path::new(path_string)) {
```

Comment:
https://rustdoc.swc.rs/swc_core/utils/swc_common/struct.SourceMap.html#method.load_file

This return type is a pointer to the source file, which has `src` https://rustdoc.swc.rs/swc_core/utils/swc_common/struct.SourceFile.html available (pub src: [Arc](https://rustdoc.swc.rs/swc_core/utils/swc_common/sync/struct.Lrc.html)<[String](https://doc.rust-lang.org/nightly/alloc/string/struct.String.html)>)

Instead of doing the redundant read below, consider extracting the spans here where we have the original source, and storing those on SourceSpecifiers instead of storing the ImportSpecifier AST nodes.